lib/ty.ml

```diff
@@ -0,0 +1,188 @@
+type _ cast =
+  | C32 : int32 cast
+  | C64 : int64 cast
+
+type sz =
+  | S32
+  | S64
+
+type t =
+  | Ty_int
+  | Ty_real
+  | Ty_bool
+  | Ty_str
+  | Ty_bitv of sz
+  | Ty_fp of sz
+
+type unop =
+  | Neg
+  | Not
+  | Clz
+  | Abs
+  | Sqrt
+  | Nearest
+  | Is_nan
+  | Ceil
+  | Floor
+  (* To remove *)
+  | Len
+  | Trim
+
+type binop =
+  | Add
+  | Sub
+  | Mul
+  | Div
+  | DivU
+  | Rem
+  | RemU
+  | Shl
+  | ShrA
+  | ShrL
+  | And
+  | Or
+  | Xor
+  | Pow
+  | Min
+  | Max
+  | Rotl
+  | Rotr
+  | Ext
+  | ExtU
+  (* To remove *)
+  | Nth
+  | Concat
+
+type relop =
+  | Eq
+  | Ne
+  | Lt
+  | LtU
+  | Gt
+  | GtU
+  | Le
+  | LeU
+  | Ge
+  | GeU
+
+type triop =
+  | Ite
+  | Substr
+
+type cvtop =
+  | ToString
+  | OfString
+  | ToBool
+  | OfBool
+  | Reinterpret_int
+  | Reinterpret_float
+  | DemoteF64
+  | PromoteF32
+  | ConvertSI32
+  | ConvertUI32
+  | ConvertSI64
+  | ConvertUI64
+  | TruncSF32
+  | TruncUF32
+  | TruncSF64
+  | TruncUF64
+  | WrapI64
+  | ExtendSI32
+  | ExtendUI32
+
+let pp_unop fmt (op : unop) =
+  match op with
+  | Neg -> Format.pp_print_string fmt "neg"
```

Comment:
Maybe putting something like:

```ocaml
open Format
let pp_string = pp_print_string
let pp = fprintf
```

at the top would wake everything that follows less dense

Reply:
Indeed, it looks much better with this